hoot-core/src/main/cpp/hoot/core/elements/Relation.cpp

```diff
@@ -144,19 +150,19 @@ Envelope* Relation::getEnvelope(const boost::shared_ptr<const ElementProvider> &
     // if any of the elements don't exist then return an empty envelope.
     if (ep->containsElement(m.getElementId()) == false)
     {
-      result->setToNull();
+      result.setToNull();
       return result;
     }
     const boost::shared_ptr<const Element> e = ep->getElement(m.getElementId());
     auto_ptr<Envelope> childEnvelope(e->getEnvelope(ep));
```

Comment:
I'll give you a dollar if you change this to a `boost::shared_ptr` instead of the depricated `auto_ptr` :dispensary:

Reply:
Tough but fair. Consider it done!